File: Iniciante/1040.c
#include <stdio.h>

int main(){

    float n1, n2, n3, n4, notaExam, media;

    scanf("%f %f %f %f", &n1, &n2, &n3, &n4);

    n1 = n1 * 2 / 10;
    n2 = n2 * 3 / 10;
    n3 = n3 * 4 / 10;
    n4 = n4 * 1 / 10;
    media = n1 + n2 + n3 + n4;

        printf("Media: %0.1f\n", media);

        if(media >= 7.0){

            printf("Aluno aprovado.\n");
        }else
        if(media >= 5.0){

            printf("Aluno em exame.\n");
            scanf("%f", &notaExam);
            printf("Nota do exame: %0.1f\n", notaExam);
            media = (media + notaExam) / 2;

            if(media >= 5.0){

                printf("Aluno aprovado.\n");

            }else   printf("Aluno reprovado.\n");

            printf("Media final: %0.1f\n", media);


        }else
            printf("Aluno reprovado.\n");


    return 0;

}

File: Iniciante/1040.java
import java.text.DecimalFormat;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        Scanner scanf = new Scanner(System.in);
        DecimalFormat dF = new DecimalFormat("0.0");

        double n1 = scanf.nextDouble();
        double n2 = scanf.nextDouble();
        double n3 = scanf.nextDouble();
        double n4 = scanf.nextDouble();

        double media = ((n1*2) + (n2*3) + (n3*4) + (n4*1)) / 10;

        System.out.println("Media: " + dF.format(media));

        if(media >= 7.0){

            System.out.println("Aluno aprovado.");
        }else
        if(media >= 5.0){

            System.out.println("Aluno em exame.");
            double notaExam = scanf.nextDouble();
            System.out.println("Nota do exame: " + dF.format(notaExam));
            media = (media + notaExam) / 2;

            if(media >= 5.0){

                System.out.println("Aluno aprovado.");

            }else System.out.println("Aluno reprovado.");

            System.out.println("Media final: " + dF.format(media));


        }else
            System.out.println("Aluno reprovado.");       
    }
}